tools/SetupFlow/DevHome.SetupFlow/ViewModels/PackageCatalogListViewModel.cs

```diff
@@ -1,100 +1,134 @@
+        IWindowsPackageManager wpm,
+        PackageCatalogViewModelFactory packageCatalogViewModelFactory)
+    {
+        _catalogDataSourceLoader = catalogDataSourceLoader;
+        _wpm = wpm;
+        _packageCatalogViewModelFactory = packageCatalogViewModelFactory;
+    }
+
+    /// <summary>
+    /// Load the package catalogs to display
+    /// </summary>
+    public async Task LoadCatalogsAsync()
+    {
+        if (!_initialized)
+        {
+            _initialized = true;
+            AddShimmers(_catalogDataSourceLoader.CatalogCount);
+            try
+            {
+                await Task.Run(async () => await _wpm.WinGetCatalog.ConnectAsync());
+                await foreach (var dataSourceCatalogs in _catalogDataSourceLoader.LoadCatalogsAsync())
+                {
+                    foreach (var catalog in dataSourceCatalogs)
+                    {
+                        var catalogVM = await Task.Run(() => _packageCatalogViewModelFactory(catalog));
+                        catalogVM.CanAddAllPackages = true;
+                        PackageCatalogs.Add(catalogVM);
+                    }
+
+                    RemoveShimmers(dataSourceCatalogs.Count);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Logger?.ReportError(Log.Component.AppManagement, $"Failed to connect to {nameof(_wpm.WinGetCatalog)}. Skipping catalogs loading operation.", e);
+            }
+
+            // Remove any remaining shimmers:
+            // This can happen if for example a catalog was detected but not
+            // displayed (e.g. catalog with no packages to display)
+            RemoveShimmers(_catalogDataSourceLoader.CatalogCount);
+        }
+    }
+
+    /// <summary>
+    /// Add package catalog shimmers
+    /// </summary>
+    /// <param name="count">Number of package catalog shimmers to add</param>
+    private void AddShimmers(int count)
+    {
+        while (count-- > 0)
+        {
+            PackageCatalogShimmers.Add(PackageCatalogShimmers.Count);
+        }
+    }
+
+    /// <summary>
+    /// Remove package catalog shimmers
+    /// </summary>
+    /// <param name="count">Number of package catalog shimmers to remove</param>
+    private void RemoveShimmers(int count)
+    {
+        while (count-- > 0 && PackageCatalogShimmers.Any())
+        {
+            PackageCatalogShimmers.Remove(PackageCatalogShimmers.Last());
+        }
+    }
+
+    [RelayCommand]
+    private void ViewAllPackages(PackageCatalogViewModel catalog)
+    {
+        AppManagementBehavior.SetHeaderVisibility(false);
+        ViewAllCatalog = catalog;
+    }
+
+    [RelayCommand]
+    private void ExitViewAllPackages()
+    {
+        AppManagementBehavior.SetHeaderVisibility(true);
+        ViewAllCatalog = null;
+    }
+
+    [RelayCommand]
+    private void OnLoaded()
+    {
+        // When the view is loaded, ensure we exit the view all packages mode
+        ExitViewAllPackages();

```

Comment:
Seems like if the user left the machine configuration page, and came back they won't be back into their original view experience right?  I'm wondering for the scenario where they clicked view all then had to scroll down, then they click on the dashboard to check something, but then come back to the machine configuration page. They'd have to click view all again and scroll down right? Mostly speaking from a user experience point of view. Not sure what the requirement was though.

Reply:
Correct, that's the behavior today if you move away from the page while in View All mode, it will reset and you have to click View All again. I added this consideration as part of the issue below, as it's technically related to it.
- https://github.com/microsoft/devhome/issues/1621